src/category_theory/arrow_class.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import category_theory.arrow
+
+/-!
+
+# Classes of arrows in a category
+
+If `C` is a category, an `arrow_class C` is a class of arrows in C. It is
+implemented here as `set (arrow C)`.
+
+If `W : arrow_class C`, and `f : X ⟶ Y` is a morphism in `C`, `W f` is the
+condition that `arrow.mk f` belongs to `W`.
+
+-/
+
+noncomputable theory
+
+namespace category_theory
+
+variables (C : Type*) [category C] {D : Type*} [category D]
+
+/-- An `arrow_class C` is a class of arrows in a category `C`. -/
+abbreviation arrow_class := set (arrow C)
```

Comment:
I'm a bit late to the party, but would it make sense to reuse [docs#category_theory.morphism_properties](https://leanprover-community.github.io/mathlib_docs/category_theory/morphism_property.html#category_theory.morphism_property) here?
Judging at the code in this PR, I would assume the switch would be quite seamless.

Reply:
Thanks, I will try to adapt to `morphism_properties`. I may have more suggestions about expanding `morphism_properties` also when I start PR-ing Quillen's homotopical algebra...